bcipy/tasks/rsvp/main_frame.py

```diff
@@ -253,3 +261,77 @@ def prepare_stimuli(self):
             timing=self.stimuli_timing,
             seq_constants=self.seq_constants)
         return stimuli
+
+
+class CriteriaEvaluator():
+    """Evaluates whether an epoch should commit to a decision based on the
+    provided criteria.
+    
+    Parameters:
+    -----------
+        continue_criteria: list of functions; if any of these evaluate to true the
+            decision maker continues.
+        commit_criteria: list of functions; if any of these return true and
+            continue_criteria are all false, decision maker commits to a decision.
+    """
+
+    def __init__(self, continue_criteria: list, commit_criteria: list):
+        self.continue_criteria = continue_criteria or []
+        self.commit_criteria = commit_criteria or []
+
+    @classmethod
+    def default(cls):
+        return cls(
+            continue_criteria=[less_than_min_iterations],
+            commit_criteria=[exceeds_max_iterations, meets_commit_threshold])
+
+    def should_commit(self, epoch: Dict, params: Dict):
+        """Evaluates the given epoch; returns true if stoppage criteria has
+        been met, otherwise false.
+
+        Parameters:
+        -----------
+            epoch - Epoch data
```

Comment:
Some explanation of what would be needed for an `epoch` may be helpful here. It looks like we're using only the`list_distribution`. If we won't need anything else, we could also have them pass that in directly.

Reply:
Our current criteria only uses list_distribution, but I want to make all the information available to future criteria. I will add some documentation to the epoch param. It should probably be an object rather than a dict at this point.